server/data/data-file/src/test/java/org/apache/james/mailrepository/FileMailRepositoryTest.java

```diff
@@ -25,32 +25,85 @@
 import org.apache.james.mailrepository.file.FileMailRepository;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
 
-public class FileMailRepositoryTest implements MailRepositoryContract {
-
-    private FileMailRepository mailRepository;
-    private MockFileSystem filesystem;
-
-    @BeforeEach
-    void init() throws Exception {
-        filesystem = new MockFileSystem();
-        mailRepository = new FileMailRepository();
-        mailRepository.setFileSystem(filesystem);
-        DefaultConfigurationBuilder defaultConfiguration = new DefaultConfigurationBuilder();
-        defaultConfiguration.addProperty("[@destinationURL]", "file://target/var/mailRepository");
-        defaultConfiguration.addProperty("[@type]", "MAIL");
-        mailRepository.configure(defaultConfiguration);
-        mailRepository.init();
+public class FileMailRepositoryTest {
+
+    public abstract class GenericFileMailRepositoryTest implements MailRepositoryContract {
+        private FileMailRepository mailRepository;
+        private MockFileSystem filesystem;
+
+        @BeforeEach
+        void init() throws Exception {
+            filesystem = new MockFileSystem();
+            mailRepository = new FileMailRepository();
+            mailRepository.setFileSystem(filesystem);
+            mailRepository.configure(getConfiguration());
+            mailRepository.init();
+        }
+
+        protected DefaultConfigurationBuilder getConfiguration() {
+            DefaultConfigurationBuilder configuration = new DefaultConfigurationBuilder();
+            configuration.addProperty("[@destinationURL]", "file://target/var/mailRepository");
+            return withConfigurationOptions(configuration);
+        }
+
+        protected abstract DefaultConfigurationBuilder withConfigurationOptions(DefaultConfigurationBuilder configuration);
+
+        @AfterEach
+        void tearDown() {
+            filesystem.clear();
+        }
+
+        @Override
+        public MailRepository retrieveRepository() {
+            return mailRepository;
+        }
+    }
+
+    @Nested
+    @DisplayName("Default configuration")
```

Comment:
My choice went to **Nested** as over generic extensions Akka `@TestTemplate` and `@ParameterizedTest` are to be specified on each test.

As I want it to be used on the entire class, this is the cleanest solution I found.

Reply:
not bad